encoder.go

```diff
@@ -102,6 +98,20 @@ func (e *encoder) setFieldByType(current reflect.Value, namespace []byte, idx in
 			return
 		}
 	}
+	{
```

Comment:
I'm not sure the extra scope is necessary.

Reply:
it is not